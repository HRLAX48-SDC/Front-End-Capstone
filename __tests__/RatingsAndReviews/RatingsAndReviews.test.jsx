import React from 'react';
import Enzyme, { shallow, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import RatingsAndReviews from '../../client/src/components/ratingandreviews/RatingAndReviews.jsx';

Enzyme.configure({ adapter: new Adapter() });

    const emptyReview = [];

test('Renders an empty review component', () =>{
    const wrapper = mount(<RatingsAndReviews ratings={emptyReview}/>);
    expect(wrapper). toContainMatchingElements(0, 'li')
})