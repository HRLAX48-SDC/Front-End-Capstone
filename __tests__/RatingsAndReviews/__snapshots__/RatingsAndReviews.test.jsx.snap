// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RatingsAndReviews /> Should correctly render 1`] = `
Array [
  <h5
    className="rating-reviews-title"
  >
    RATINGS AND REVIEWS
  </h5>,
  <div
    id="rating-reviews-main"
  >
    <div
      className="hstack gap-3"
    >
      <div
        className="card"
        style={
          Object {
            "width": "26rem",
          }
        }
      >
        <div
          className="card"
        >
          <div
            className="card-title h5"
          />
          <div
            className="card-body"
          >
            <div
              className="card"
            >
              <div
                className="card-title h5"
              >
                <div
                  className="hstack gap-3"
                >
                  <div
                    className="average-review-header"
                  >
                    0.0
                  </div>
                  <span
                    style={
                      Object {
                        "direction": "ltr",
                        "display": "inline-block",
                        "touchAction": "none",
                      }
                    }
                  >
                    <span
                      aria-hidden="true"
                      className="react-simple-star-rating"
                      style={
                        Object {
                          "cursor": "",
                          "display": "inline-block",
                          "overflow": "hidden",
                          "position": "relative",
                          "userSelect": "none",
                          "verticalAlign": "middle",
                          "whiteSpace": "nowrap",
                        }
                      }
                    >
                      <span
                        className="empty-icons"
                        style={
                          Object {
                            "color": "#cccccc",
                            "display": "inline-block",
                          }
                        }
                      >
                        <svg
                          className="star-svg"
                          fill="currentColor"
                          height={35}
                          style={Object {}}
                          viewBox="0 0 24 24"
                          width={35}
                        >
                          <path
                            d="M12,17.27L18.18,21l-1.64-7.03L22,9.24l-7.19-0.61L12,2L9.19,8.63L2,9.24l5.46,4.73L5.82,21L12,17.27z"
                            fill="currentColor"
                            stroke="none"
                            strokeMiterlimit="10"
                            strokeWidth={0}
                          />
                        </svg>
                        <svg
                          className="star-svg"
                          fill="currentColor"
                          height={35}
                          style={Object {}}
                          viewBox="0 0 24 24"
                          width={35}
                        >
                          <path
                            d="M12,17.27L18.18,21l-1.64-7.03L22,9.24l-7.19-0.61L12,2L9.19,8.63L2,9.24l5.46,4.73L5.82,21L12,17.27z"
                            fill="currentColor"
                            stroke="none"
                            strokeMiterlimit="10"
                            strokeWidth={0}
                          />
                        </svg>
                        <svg
                          className="star-svg"
                          fill="currentColor"
                          height={35}
                          style={Object {}}
                          viewBox="0 0 24 24"
                          width={35}
                        >
                          <path
                            d="M12,17.27L18.18,21l-1.64-7.03L22,9.24l-7.19-0.61L12,2L9.19,8.63L2,9.24l5.46,4.73L5.82,21L12,17.27z"
                            fill="currentColor"
                            stroke="none"
                            strokeMiterlimit="10"
                            strokeWidth={0}
                          />
                        </svg>
                        <svg
                          className="star-svg"
                          fill="currentColor"
                          height={35}
                          style={Object {}}
                          viewBox="0 0 24 24"
                          width={35}
                        >
                          <path
                            d="M12,17.27L18.18,21l-1.64-7.03L22,9.24l-7.19-0.61L12,2L9.19,8.63L2,9.24l5.46,4.73L5.82,21L12,17.27z"
                            fill="currentColor"
                            stroke="none"
                            strokeMiterlimit="10"
                            strokeWidth={0}
                          />
                        </svg>
                        <svg
                          className="star-svg"
                          fill="currentColor"
                          height={35}
                          style={Object {}}
                          viewBox="0 0 24 24"
                          width={35}
                        >
                          <path
                            d="M12,17.27L18.18,21l-1.64-7.03L22,9.24l-7.19-0.61L12,2L9.19,8.63L2,9.24l5.46,4.73L5.82,21L12,17.27z"
                            fill="currentColor"
                            stroke="none"
                            strokeMiterlimit="10"
                            strokeWidth={0}
                          />
                        </svg>
                      </span>
                      <span
                        className="filled-icons"
                        style={
                          Object {
                            "color": "#ffbc0b",
                            "display": "inline-block",
                            "left": 0,
                            "overflow": "hidden",
                            "position": "absolute",
                            "top": 0,
                            "transition": "",
                            "whiteSpace": "nowrap",
                            "width": "0%",
                          }
                        }
                        title="0 out of 5"
                      >
                        <svg
                          className="star-svg"
                          fill="currentColor"
                          height={35}
                          style={Object {}}
                          viewBox="0 0 24 24"
                          width={35}
                        >
                          <path
                            d="M12,17.27L18.18,21l-1.64-7.03L22,9.24l-7.19-0.61L12,2L9.19,8.63L2,9.24l5.46,4.73L5.82,21L12,17.27z"
                            fill="currentColor"
                            stroke="none"
                            strokeMiterlimit="10"
                            strokeWidth={0}
                          />
                        </svg>
                        <svg
                          className="star-svg"
                          fill="currentColor"
                          height={35}
                          style={Object {}}
                          viewBox="0 0 24 24"
                          width={35}
                        >
                          <path
                            d="M12,17.27L18.18,21l-1.64-7.03L22,9.24l-7.19-0.61L12,2L9.19,8.63L2,9.24l5.46,4.73L5.82,21L12,17.27z"
                            fill="currentColor"
                            stroke="none"
                            strokeMiterlimit="10"
                            strokeWidth={0}
                          />
                        </svg>
                        <svg
                          className="star-svg"
                          fill="currentColor"
                          height={35}
                          style={Object {}}
                          viewBox="0 0 24 24"
                          width={35}
                        >
                          <path
                            d="M12,17.27L18.18,21l-1.64-7.03L22,9.24l-7.19-0.61L12,2L9.19,8.63L2,9.24l5.46,4.73L5.82,21L12,17.27z"
                            fill="currentColor"
                            stroke="none"
                            strokeMiterlimit="10"
                            strokeWidth={0}
                          />
                        </svg>
                        <svg
                          className="star-svg"
                          fill="currentColor"
                          height={35}
                          style={Object {}}
                          viewBox="0 0 24 24"
                          width={35}
                        >
                          <path
                            d="M12,17.27L18.18,21l-1.64-7.03L22,9.24l-7.19-0.61L12,2L9.19,8.63L2,9.24l5.46,4.73L5.82,21L12,17.27z"
                            fill="currentColor"
                            stroke="none"
                            strokeMiterlimit="10"
                            strokeWidth={0}
                          />
                        </svg>
                        <svg
                          className="star-svg"
                          fill="currentColor"
                          height={35}
                          style={Object {}}
                          viewBox="0 0 24 24"
                          width={35}
                        >
                          <path
                            d="M12,17.27L18.18,21l-1.64-7.03L22,9.24l-7.19-0.61L12,2L9.19,8.63L2,9.24l5.46,4.73L5.82,21L12,17.27z"
                            fill="currentColor"
                            stroke="none"
                            strokeMiterlimit="10"
                            strokeWidth={0}
                          />
                        </svg>
                      </span>
                    </span>
                  </span>
                </div>
                <h5
                  className="total-ratings-header"
                >
                   ratings
                </h5>
              </div>
              <div
                className="card-body"
              >
                <div
                  className="spinner-border"
                  role="status"
                >
                  <span
                    className="visually-hidden"
                  >
                    Loading...
                  </span>
                </div>
              </div>
            </div>
            <div
              className="card"
            >
              <div
                className="card-body"
              >
                <div
                  className="spinner-border"
                  role="status"
                >
                  <span
                    className="visually-hidden"
                  >
                    Loading...
                  </span>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="card"
        style={
          Object {
            "width": "55rem",
          }
        }
      >
        <div
          className="card"
        >
          <div
            className="reviews-dropdown-stack hstack gap-2"
          >
            <div>
              Displaying
            </div>
            <b>
              0
               / 
              0
            </b>
            <div>
              reviews, sorted by:
            </div>
            <div
              className="dropdown"
            >
              <button
                aria-expanded={false}
                className="dropdown-toggle btn btn-outline-secondary"
                disabled={false}
                id="dropdown-basic-button"
                onClick={[Function]}
                type="button"
              >
                relevant
              </button>
            </div>
            <div
              className="ms-auto"
            >
              <div>
                <button
                  className="open-modal btn btn-outline-secondary"
                  disabled={false}
                  onClick={[Function]}
                  type="button"
                >
                  Submit Review
                </button>
              </div>
            </div>
          </div>
          <div
            className="search-bar"
          >
            <form
              className=""
            >
              <div
                className="input-group"
              >
                <span
                  className="input-group-text"
                >
                  <i
                    className="fas fa-search"
                  />
                </span>
                <input
                  className="form-control"
                  onChange={[Function]}
                  placeholder="Search reviews..."
                  type="text"
                  value=""
                />
              </div>
            </form>
          </div>
        </div>
        <div
          className="reviews-list card"
        >
          <div
            className="card-body"
          >
            <div
              className="vstack gap-3"
            >
              <button
                className="show-less btn btn-outline-secondary"
                disabled={false}
                onClick={[Function]}
                type="button"
              >
                Collapse
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
]
`;
